cdap-api/src/main/java/co/cask/cdap/internal/schedule/TimeSchedule.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.schedule;
+
+import co.cask.cdap.api.schedule.Schedule;
+
+/**
+ * Defines a cron-based schedule for running a program.
+ */
+public final class TimeSchedule extends Schedule {
+
+  private final String cronExpression;
+
+  public TimeSchedule(String name, String description, String cronExpression) {
+    super(name, description);
+    this.cronExpression = cronExpression;
+  }
+
+  /**
+   * @return Cron expression for the schedule, if this schedule is a time based schedule.
+   */
+  public String getCronExpression() {
```

Comment:
Ok, I will do that. But I still need to have another ``cronExpression`` field in ``TimeSchedule``, right?